app/controllers/comments_controller.rb

```diff
@@ -1,32 +1,31 @@
 class CommentsController < ApplicationController
-  before_action :find_article, only: %i[create]
   before_action :find_comment, only: %i[destroy]
-
   http_basic_authenticate_with name: 'user', password: 'secret', only: :destroy
 
   def create
-    @comment = @article.comments.create(comments_params)
-    redirect_to article_path(@article)
+    @comment = Comment.new(comments_params)
+    @comment.article_id = params[:article_id]
+    begin
+      @comment.save!
+    rescue ActiveRecord::RecordInvalid
+    end
+    redirect_back fallback_location: articles_url
```

Comment:
This should be inside the `begin` block, like 

```ruby
    begin
      @comment.save!
       redirect_back fallback_location: articles_url
    rescue ActiveRecord::RecordInvalid
    end

```

Reply:
Update of `create` method on `CommentsController`
```ruby
def create
    @comment = Comment.new(comment_params)
    @comment.save
    redirect_article
end
```